### ISSUE TITLE
Problems with xml encoding of predefined entities in element names (artists, titles, albums etc) during ContentDirectory1#Browse

### ISSUE THREAD

**Issue by Nadahar** [UniversalMediaServer/UniversalMediaServer#579]
I started this "case" on your [help forum](http://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=3496&p=16679#p16679). My original problem was that some music titles and artists names where shown wrong to the extent that it's hard to read. Since then I've studied UMS, Java, Eclipse, and Maven much more than I'd want to :open_mouth:

**Comment by Nadahar** [comment]
I even got @valib to commit 2258692a944a7f86874b12ec6846d2f6b5d5118f and then later revert ab04b6cede2085c9854ac9354637cb3c8805427b. I'm sorry about that, it seemed like an obvious bug AND it seemed to work on my Onkyo renderer (even though I've figured out that it's not that simple now).

**Comment by Nadahar** [comment]
I just did a Wireshark trace for Plex as well, and it has the same behaviour as UMS - and is not presenting titles etc. correctly on the Onkyo:
```
&lt;dc:title&gt;Konstant i Opel &amp;apos;34&amp;apos; ds &amp;quot;3&amp;amp;&amp;quot; &amp;gt;&amp;gt; ds&lt;/dc:title&gt;
```

**Comment by Nadahar** [comment]
After much looking, I finally found that [UPnP-av-ContentDirectory-v1-Service] (http://upnp.org/specs/av/UPnP-av-ContentDirectory-v1-Service.pdf) (which UMS uses) refers to [XML Section 2.4 Character Data and Markup] (http://www.w3.org/TR/REC-xml/#syntax):
> Note that since the DIDL-Lite format of Result is based on XML, it needs to be escaped (using the normal XML rules: [XML] Section 2.4 Character Data and Markup) before embedding in a SOAP response message. In addition, when a variable of type A_ARG_TYPE_Result is employed as a parameter in a comma-separated list, commas (‘,’) that appear within XML CDATA must be escaped (as \,), see section 2.3.1.

[XML Section 2.4 Character Data and Markup] (http://www.w3.org/TR/REC-xml/#syntax) states:
> Text consists of intermingled character data and markup. [Definition: Markup takes the form of start-tags, end-tags, empty-element tags, entity references, character references, comments, CDATA section delimiters, document type declarations, processing instructions, XML declarations, text declarations, and any white space that is at the top level of the document entity (that is, outside the document element and not inside any other markup).]
> 
> [Definition: All text that is not markup constitutes the character data of the document.]
> 
> The ampersand character (&) and the left angle bracket (<) MUST NOT appear in their literal form, except when used as markup delimiters, or within a comment, a processing instruction, or a CDATA section. If they are needed elsewhere, they MUST be escaped using either numeric character references or the strings " \&amp; " and " \&lt; " respectively. The right angle bracket (\>) may be represented using the string " \&gt; ", and MUST, for compatibility, be escaped using either " \&gt; " or a character reference when it appears in the string " ]]> " in content, when that string is not marking the end of a CDATA section.
> 
> In the content of elements, character data is any string of characters which does not contain the start-delimiter of any markup and does not include the CDATA-section-close delimiter, " \]]> ". In a CDATA section, character data is any string of characters not including the CDATA-section-close delimiter, " \]]> ".
> 
> To allow attribute values to contain both single and double quotes, the apostrophe or single-quote character (') **may** be represented as " \&apos; ", and the double-quote character (") as " \&quot; ".

My conclusion is then that is should be safe not to escape single- and double-quote characters since escaping them is optional and not all renderers expect them to be encoded.

**Comment by Nadahar** [comment]
This is fixed with #592.
